### PR TITLE
build optimize

### PR DESCRIPTION
**Pull request by yshysh**

### CODE REVIEW

File: cmake/hdfs.cmake [hunk 1]
```diff
@@ -8,7 +8,7 @@ ExternalProject_Add(
     extern_hdfs
     DOWNLOAD_DIR ${THIRD_PARTY_PATH}
     DOWNLOAD_COMMAND rm -rf ${HDFS_SOURCES_DIR} && git clone https://github.com/acmol/pivotalrd-libhdfs3.git
-    CONFIGURE_COMMAND cd ${HDFS_SOURCES_DIR} && mkdir -p build && cd build && ../bootstrap --prefix=${HDFS_INSTALL_DIR} --dependency=${UUID_INSTALL_DIR}:${PROTOBUF_INSTALL_DIR}:${LIBGSASL_INSTALL_DIR}:${LIBXML2_INSTALL_DIR}:${KRB_INSTALL_DIR}
+    CONFIGURE_COMMAND cd ${HDFS_SOURCES_DIR} && mkdir -p build && cd build && CC=gcc CXX=g++ ../bootstrap --prefix=${HDFS_INSTALL_DIR} --dependency=${UUID_INSTALL_DIR}:${PROTOBUF_INSTALL_DIR}:${LIBGSASL_INSTALL_DIR}:${LIBXML2_INSTALL_DIR}:${KRB_INSTALL_DIR}
```
Comment (advancedxy) [inline cmake/hdfs.cmake:11]: Why CC and CXX are needed?

File: build_support/build_deps.sh [hunk 2]
```diff
@@ -8,14 +8,20 @@ if [ -e /.dockerenv ];then
...
 tar zxf cmake-3.10.0-Linux-x86_64.tar.gz
 
 echo 'export PATH='$PWD'/cmake-3.10.0-Linux-x86_64/bin/:$PATH' >> profiles/cmake.profile
 source ./profiles/cmake.profile
 
+wget https://www.samba.org/ftp/ccache/ccache-3.3.4.tar.gz -O ccache-3.3.4.tar.gz
+tar zxvf ccache-3.3.4.tar.gz
+pushd ccache-3.3.4
+./configure && make && make install
+popd
```
Comment (advancedxy) [inline build_support/build_deps.sh:23]: Isn't ccache already in CentOS's yum repo?

Comment (yshysh) [reply build_support/build_deps.sh:23]: yes

File: cmake/hdfs.cmake (hunk 1 above)
Comment (yshysh) [reply cmake/hdfs.cmake:11]: so many libs, can not make sure when CC is needed or CXX is needed

File: build_support/build_deps.sh (hunk 2 above)
Comment (acmol) [reply build_support/build_deps.sh:23]: If it is in the yum repo, why don't you just install it by using yum?

File: cmake/hdfs.cmake (hunk 1 above)
Comment (acmol) [reply cmake/hdfs.cmake:11]: @yshysh I think you did not answer correctly  about what @advancedxy asked.

File: build_support/build_deps.sh (hunk 2 above)
Comment (yshysh) [reply build_support/build_deps.sh:23]: no, I say yum repo can not find ccache.

File: cmake/hdfs.cmake (hunk 1 above)
Comment (yshysh) [reply cmake/hdfs.cmake:11]: CC and CXX are needed for build parameter that used to optimizing build.

**Pull request closed without merging**